src/main/java/randoop/sequence/VariableRenamer.java

```diff
@@ -61,49 +69,103 @@ static String getVariableName(Type type) {
       if (type.equals(JavaTypes.BYTE_TYPE)) {
         return "byte";
       }
-      // otherwise, use the first character of the type name
+      if (type.equals(JavaTypes.INT_TYPE)) {
+        return "int";
+      }
+      if (type.equals(JavaTypes.BOOLEAN_TYPE)) {
+        return "bool";
+      }
+      if (type.equals(JavaTypes.FLOAT_TYPE)) {
+        return "float";
+      }
+      if (type.equals(JavaTypes.DOUBLE_TYPE)) {
+        return "double";
+      }
+      if (type.equals(JavaTypes.SHORT_TYPE)) {
+        return "short";
+      }
+
+      // Otherwise, use the first character of the type name.
       return type.getName().substring(0, 1);
     }
 
-    // special cases:  Object, String, Class
+    // Special cases:  Object, String, Class.
     if (type.isObject()) {
       return "obj";
     } else if (type.isString()) {
       return "str";
     } else if (type.equals(JavaTypes.CLASS_TYPE)) {
       return "cls";
     }
-    // TODO: add more special cases:
-    //  * if it implements List, use "list"
-    //  * if it implements Iterator, use "itor"
-    // ... more?
 
     if (type.isParameterized()) {
       ClassOrInterfaceType classType = (ClassOrInterfaceType) type;
       String varName = classType.getSimpleName().toLowerCase();
       if (varName.equals("class")) {
         return "cls";
       }
-      for (TypeArgument argument : classType.getTypeArguments()) {
+
+      // Special cases for parameterized types.
+      if (varName.contains("iterator")) {
+        // Iterator takes precedence, in cases like ListIterator.
+        varName = "itor";
+      } else if (varName.contains("list")) {
+        // List comes before array, in cases like ArrayList.
+        varName = "list";
+      } else if (varName.contains("set")) {
+        varName = "set";
+      } else if (varName.contains("map")) {
+        varName = "map";
+      } else if (varName.contains("queue")) {
+        varName = "queue";
+      } else if (varName.contains("collection")) {
+        varName = "collection";
+      } else if (varName.contains("array")) {
+        varName = "array";
+      }
+
+      if (classType.getTypeArguments().size() >= 1) {
+        TypeArgument argument = classType.getTypeArguments().get(0);
+
         if (argument.isWildcard()) {
-          varName += "_" + "wildcard";
+          // Capitalize the variable name while preserving any capitalized letters after the first letter.
```

Comment:
This is a good idea.  But wasn't varName already set to all lowercase above?  The change I suggested above will let you fix this too.